examples/SerialIface/vgm.py

```diff
@@ -32,21 +36,18 @@ def play(opl, vgm_stream):
       delay_us = 0
     elif DELAY_N == opcode:
       delay_samples, = struct.unpack_from('H', vgm_stream.read(2))
-      delay_us = _samples_to_us(delay_samples)
+      delay_us += _samples_to_us(delay_samples)
     elif DELAY_735 == opcode:
-      delay_us = _samples_to_us(735)
+      delay_us += _samples_to_us(735)
     elif DELAY_882 == opcode:
-      delay_us = _samples_to_us(882)
-    elif opcode >= DELAY_N1:
+      delay_us += _samples_to_us(882)
+    elif DELAY_N1 == opcode & 0xf0:
       delay_samples = 1 + int(opcode & 0xf)
-      delay_us = _samples_to_us(delay_samples)
+      delay_us += _samples_to_us(delay_samples)
     elif opcode == END_DATA:
       break
-    elif opcode >= 0x51 and opcode <= 0x5f:
-      # opcodes for writing to other YMF chips, skip
-      vgm_stream(2)
-    elif opcode == 0:
-      pass
+    elif (opcode & 0xf0) in (YM_CHIP_1, YM_CHIP_2):
+      vgm_stream.read(2 if (opcode & 0x0f) else 1)
     else:
```

Comment:
I'm having some hardware problems at the moment to really verify, but it looks like this reintroduces the issue where 0x00 opcodes are throwing an exception.

Reply:
Yep looks like it was due to some other error. Zero Wing and some others I tried play fine. Thanks for the update!